pootle/apps/pootle_translationproject/views.py

```diff
@@ -240,15 +239,19 @@ def get_context_data(self, *args, **kwargs):
             {'parent': get_parent(self.object)})
         return ctx
 
-    def get_upload_widget(self):
-        ctx = {}
-        has_upload = (
+    @property
+    def can_upload(self):
+        return (
             "import_export" in settings.INSTALLED_APPS
             and self.request.user.is_authenticated()
-            and check_permission('translate', self.request))
-        if has_upload:
-            if "po" in self.project.filetype_tool.valid_extensions:
-                ctx.update(handle_upload_form(self.request, self.tp))
+            and (self.request.user.is_superuser
+                 or "translate" in self.request.permissions
```

Comment:
user with translate permissions only shouldn't be able to upload as different user. it looks like you don't differ translate and administrate here.